### ISSUE TITLE
Update web interface for 4.0 release

### ISSUE THREAD

**Issue by jakubpawlowicz** [jakubpawlowicz/clean-css#858]
It's not in the 4.0 milestone as we need the release itself to be able to serve the web-compatible build. It should be ready for 4.0 release though.

Does #856 mean we'd need to pull in promises polyfill for those browsers which doesn't support it?
http://caniuse.com/#feat=promises
https://cdn.polyfill.io/v2/docs/features/#Promise

**Comment by alexlamsl** [comment]
So is the old `minify()` completely removed now, and everything needs to be done via promise?

Given that nothing in the code is currently asynchronous this seems a bit excessive, not to mention compatibility concerns.

**Comment by jakubpawlowicz** [comment]
The `minify` method will still be there, but will always return a promise. Currently if want to fetch remote `@import`s the code is already asynchronous.

Here's an example of using Promises:

```js
new CleanCSS(options)
  .minify(input)
  .then(function (output) {
    // output.styles and everything else available as now
  })
  .catch(function (error) {
     // any errors
   });
```

I'm still not sold to switching to promises now, but number of browsers not supporting them is dwindling.

**Comment by alexlamsl** [comment]
Ah yes, `@import` functionality currently requires `callback` indeed.

The way I see it, the current sync/callback interface can be wrapped around by a `Promise` in supported browsers. However, if `minify()` is hardcoded to return a `Promise`, then the only way out is some form of `es-shim`, plus all the user code will have to deal with that new language feature.

May be `options.usePromise` as an experimental flag and see how widespread the adoption might get?

**Comment by jakubpawlowicz** [comment]
@alexlamsl correct me if I'm wrong but promises are just an API not a language feature and a polyfill is available for IE9+.

Also traffic stats from https://jakubpawlowicz.github.io/clean-css/ tell me that over 98% of browsers used there already support promises, the only exception being IE11.

Of course there is a question of using it within browser on other sites.

**Comment by alexlamsl** [comment]
@jakubpawlowicz indeed, I confused Promise with `await` etc. as they are often mentioned together.

As an interesting side note, AngularJS has their own implementation of `$q` and has yet to make the move onto native Promise API.

**Comment by jakubpawlowicz** [comment]
^ looks like that's it - will be merged once 4.0 is out

**Comment by jakubpawlowicz** [comment]
Ready. 👍
